app/controllers/application_controller.rb

```diff
@@ -383,6 +384,13 @@ def after_sign_in_path_for(resourse)
     end
   end
 
+  def set_display_expiration_notice
+    ext_service_active = PlanService::API::Api.plans.active?
+    is_expired = Maybe(@current_plan)[:expired].or_else(false)
```

Comment:
Did you check that `@current_plan` can be `nil`? If yes, then wrapping in Maybe is necessary. If it's always non-nil, then I think the Maybe wrapping should be removed.